Add Unit Test

winspiker opened the pull request:
Refactor project &
add UnitTests to some classes

winspiker committed:
Add Unit Test
Refactor project &
add UnitTests to some classes

loperd commented on tests/Address/ServerAddressResolverTest.php:
```diff
@@ -0,0 +1,41 @@
+    }
+
+
+    public function test_failed_resolve_ip_address(): void  // ?????
```
????

loperd commented on tests/Address/ServerAddressResolverTest.php:
```diff
@@ -0,0 +1,41 @@
+    public function serverAddressProvider(): array
+    {
+        return [
+            ['a4craft.top', '51.91.214.17'],
```
ip поменяется и тест перестанет быть валидным))

loperd commented on tests/Service/VarTypeFactoryTest.php:
```diff
@@ -0,0 +1,40 @@
+    {
+        $buffer = VarTypeFactory::createVarInt($randomInt);
+
+        self::assertEquals($buffer->readUint8(), $randomInt);
```
забавно. и этот тест проходит?

winspiker committed:
Add Packet tests

Took 4 hours 34 minutes

Took 7 minutes

Took 1 hour 19 minutes

Took 6 minutes

Took 1 minute

loperd commented on .gitignore:
```diff
@@ -1,3 +1,5 @@
-.php-cs-fixer.cache
\ No newline at end of file
+.php-cs-fixer.cache
+test.php
```
rm

loperd commented on tests/Address/ServerAddressTest.php:
```diff
@@ -0,0 +1,70 @@
+    public function serverAddressProvider(): array
+    {
+        return [
+          ['a4craft.top', '51.91.214.17', null, '51.91.214.17:25565'],
```
replace a4craft.top with example.org

loperd commented on tests/Address/ServerAddressResolverTest.php:
```diff
@@ -0,0 +1,35 @@
+        return [
+            ['a4craft top'],
+            ['65.109.108.27.0'],
+            ['mc.playmine$org'],
```
replace to examples

loperd commented on tests/Common/Query/Packet/FullStatPacketTest.php:
```diff
@@ -0,0 +1,149 @@
+                    "port" => 25565,
+                    "host" => "127.0.1.1",
+                    "players" => [
+                        0 => '',
```
it is incorrect

loperd commented on tests/Common/Query/Packet/FullStatPacketTest.php:
```diff
@@ -0,0 +1,149 @@
+//    }
+//
+//    public function packetFakeDataProvider(): array
+//    {
```
remove or rewrite

loperd commented on tests/Helpers/PacketFactory.php:
```diff
@@ -0,0 +1,77 @@
+{
+    public static function createBasicStatBuffer(string $motd, string $smp, string $world, string $numPlayers, string $maxPlayers, string $host = '127.0.1.1'): ByteBuffer
+    {
+
```
remove empty spaces

loperd commented on tests/Helpers/PacketFactory.php:
```diff
@@ -0,0 +1,77 @@
+//        $buffer->append("\x0");
+//        $buffer->append($plugins);
+//        $buffer->append("\x0");
+//        $buffer->append($map);
```
remove dead code

loperd commented on src/Loper/MinecraftQueryClient/Common/Query/Packet/FullStatPacket.php:
```diff
@@ -75,7 +75,10 @@ public function read(InputStream $is, ProtocolVersion $protocol): void
         );
-        $this->players = \explode("&#0;", $players);
+        if ('' !== $players) {
+            $this->players = \explode("&#0;", $players);
```
Remove if and use ternary operator:

```php
$this->players = '' === $players ? [] : \explode("&#0;", $players);
```
